common/Geometry2d/Line.hpp

```diff
@@ -5,20 +5,19 @@
 
 namespace Geometry2d {
 class Circle;
+class Segment;
 
 class Line {
 public:
     Line() {}
 
-    Line(const Line& other) {
-        pt[0] = other.pt[0];
-        pt[1] = other.pt[1];
+    explicit Line(Point p1, Point p2) : pt{p1, p2} {
+        // assert(p1 != p2);
     }
 
-    Line(Point p1, Point p2) {
-        pt[0] = p1;
-        pt[1] = p2;
-    }
+    explicit Line(const Line& other) : Line(other.pt[0], other.pt[1]) {}
```

Comment:
no. Not really a reason to even have an implementation of the implicit copy constructor really.